### PR TITLE
WIP: Update mappings

### PR DESCRIPTION
**Pull request by jesper-friis**
Updated the way to calculate a mapping route.

mapping_route() must be tested! It contains most likely bugs...

make_instance() should be reimplemented to use the new mapping mapping_route()

### CODE REVIEW

File: bindings/python/mappings.py [hunk 1]
```diff
@@ -32,81 +33,462 @@ class InconsistentTriplesError(MappingError):
...
+        triples: Sequence of (subject, steptype, object) triples.
+            It is safe to pass a generator expression too.
+        mapsTo: How 'mapsTo' is written in `triples`.
+        subClassOf: How 'subClassOf' is written in `triples`.  Set it
+            to None if subclasses should not be considered.
+        function_mappers: Sequence of mapping functions that takes `triples`
+            as argument and return a dict mapping output IRIs to a list
+            of `(function_iri, [input_iris, ...])` tuples.
+        hasName: How 'hasName' is written in `triples`.  Used for naming
+            function in put parameters.  The default is to use rdfs:label.
```
Comment (francescalb) [inline bindings/python/mappings.py:337]: Does it check all rdfs:label? And also, can we ask for rdfs:label in one ontology and skos:prefLabel in another?

File: bindings/python/mappings.py [hunk 2]
```diff
@@ -32,81 +33,462 @@ class InconsistentTriplesError(MappingError):
     """Inconsistcy in RDF triples."""
 
 
+class MissingRelationError(MappingError):
```
Comment (francescalb) [inline bindings/python/mappings.py:36]: Make Mapping a DliteException or DliteError

File: bindings/python/mappings.py (hunk 1 above)
Comment (jesper-friis) [reply bindings/python/mappings.py]: The argument has been updated to `label`. It checks all rdfs:label in the ontology.

It does not support asking for rdfs:label in one ontology and skos:prefLabel in another, hence mapping functions must be labeled consistently.

File: bindings/python/mappings.py (hunk 2 above)
Comment (jesper-friis) [reply bindings/python/mappings.py:55]: Fixed

**Comment by jesper-friis** [comment]
Note that this branch has been merged into branch "mappings-triplestore". See PR #284 

It should be safe to close this PR and continue with PR #284

**Comment by jesper-friis** [comment]
Replaced by PR #284

**Pull request closed without merging**